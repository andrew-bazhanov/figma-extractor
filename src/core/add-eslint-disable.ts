const maxLineRuleComment = '/* eslint-disable max-lines */';

const namingConventionRuleComment = '/* eslint-disable @typescript-eslint/naming-convention */';

export const addEslintDisableAtTheTopOfText = (
  content: string,
  options: {
    disabledMaxLines: boolean;
    disabledTypescriptNamingConvention: boolean;
  },
) => {
  return `
    ${options.disabledMaxLines ? maxLineRuleComment : ''}
    ${options.disabledTypescriptNamingConvention ? namingConventionRuleComment : ''}
    ${content}
    `;
};
