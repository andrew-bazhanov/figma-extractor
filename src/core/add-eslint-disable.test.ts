import { addEslintDisableAtTheTopOfText } from './add-eslint-disable';

describe('addEslintDisableAtTheTopOfText', () => {
  it('should return content with all rules', () => {
    expect(
      addEslintDisableAtTheTopOfText(
        `any content
    any content
    and more content
    `,
        {
          disabledMaxLines: true,
          disabledTypescriptNamingConvention: true,
        },
      ),
    ).toMatchSnapshot();
  });

  it('should return content with only max-line rule', () => {
    expect(
      addEslintDisableAtTheTopOfText(
        `any content
    any content
    and more content
    `,
        {
          disabledMaxLines: true,
          disabledTypescriptNamingConvention: false,
        },
      ),
    ).toMatchSnapshot();
  });

  it('should return content with only naming-convention rule', () => {
    expect(
      addEslintDisableAtTheTopOfText(
        `any content
    any content
    and more content
    `,
        {
          disabledMaxLines: false,
          disabledTypescriptNamingConvention: true,
        },
      ),
    ).toMatchSnapshot();
  });

  it('should return content without any rules', () => {
    expect(
      addEslintDisableAtTheTopOfText(
        `any content
    any content
    and more content
    `,
        {
          disabledMaxLines: false,
          disabledTypescriptNamingConvention: false,
        },
      ),
    ).toMatchSnapshot();
  });
});
