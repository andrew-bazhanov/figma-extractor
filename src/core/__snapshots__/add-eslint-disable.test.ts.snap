// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addEslintDisableAtTheTopOfText should return content with all rules 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    any content
    any content
    and more content
    
    "
`;

exports[`addEslintDisableAtTheTopOfText should return content with only max-line rule 1`] = `
"
    /* eslint-disable max-lines */
    
    any content
    any content
    and more content
    
    "
`;

exports[`addEslintDisableAtTheTopOfText should return content with only naming-convention rule 1`] = `
"
    
    /* eslint-disable @typescript-eslint/naming-convention */
    any content
    any content
    and more content
    
    "
`;

exports[`addEslintDisableAtTheTopOfText should return content without any rules 1`] = `
"
    
    
    any content
    any content
    and more content
    
    "
`;
